R/get.R

```diff
@@ -0,0 +1,597 @@
+
+
+#' Get user information from the RStudio Connect server
+#' 
+#' @param src The source object
+#' @param page_size the number of records to return per page (max 500)
+#' @param page_number the page number you wish to query
+#' @param prefix Filters users by prefix (username, first name, or last name). 
+#' The filter is case insensitive.
+#' 
+#' @return 
+#' A tibble with the following columns:
+#' \itemize{
+#'   \item{\strong{email}}{The user's email}
+#'   \item{\strong{username}}{The user's username}
+#'   \item{\strong{first_name}}{The user's first name}
+#'   \item{\strong{last_name}}{The user's last name}
+#'   \item{\strong{user_role}}{The user's role. It may have a value of 
+#'   administrator, publisher or viewer.}
+#'   \item{\strong{created_time}}{The timestamp (in RFC3339 format) when the 
+#'   user was created in the RStudio Connect server}
+#'   \item{\strong{updated_time}}{The timestamp (in RFC3339 format) when the 
+#'   user was last updated in the RStudio Connect server}
+#'   \item{\strong{active_time}}{The timestamp (in RFC3339 format) when the 
+#'   user was last active on the RStudio Connect server}
+#'   \item{\strong{confirmed}}{When false, the created user must confirm their 
+#'   account through an email. This feature is unique to password 
+#'   authentication.}
+#'   \item{\strong{locked}}{Whether or not the user is locked}
+#'   \item{\strong{guid}}{The user's GUID, or unique identifier, in UUID RFC4122 format}
+#' }
+#' 
+#' @details 
+#' Please see https://docs.rstudio.com/connect/api/#getUsers for more information 
+#' 
+#' @examples 
+#' \dontrun{
+#'   library(connectapi)
+#'   client <- connect()
+#'   
+#'   # get the first 20 users
+#'   get_users(client, page_size = 20)
+#'   
+#'   # get the second twenty users
+#'   get_users(client, page_size = 20, page_number = 2)
+#' 
+#' }
+#' 
+#' @export
+get_users <- function(src, page_size = 20, page_number = 1, prefix = NULL){
+  validate_R6_class("Connect", src)
+  
+  res <- src$users(
+    page_size = page_size,
+    page_number = page_number,
+    prefix = prefix
+  )
+  
+  ## TODO: Add a pagination option for users? This could be done using the 
+  ## res$total value and changing the page_size and page_number arguments
+  ## with a limit arguemnt. Then a while loop could be written similar 
+  ## to page_cursor
```

Comment:
Yes! I totally agree. I think we will ultimately want to write another `page_` helper that can simplify this pattern / allow for re-use across the different API endpoints that use this type of paging. I haven't had pressing need for this since I have not had need to worry about > 400 users or groups 😅

Reply:
Same here. I didn't tackle it because I also don't have a need for it at the moment (and may never). But in theory it  doesn't sound overly difficult since the overall structure can be modeled off the other while loops, so I figured I would leave the comment since it was something I saw you were doing in other places